kasa/smartdevice.py

```diff
@@ -793,19 +818,17 @@ async def connect(
         The device type is discovered by querying the device.
 
         :param host: Hostname of device to query
-        :param device_type: Device type to use for the device.
-            If not given, the device type is discovered by querying the device.
-            If the device type is already known, it is preferred to pass it
-            to avoid the extra query to the device to discover its type.
+        :param config: Connection parameters to ensure the correct protocol
+            and connection options are used.
         :rtype: SmartDevice
         :return: Object for querying/controlling found device.
         """
         from .device_factory import connect  # pylint: disable=import-outside-toplevel
 
-        return await connect(
-            host=host,
-            port=port,
-            timeout=timeout,
-            credentials=credentials,
-            device_type=device_type,
-        )
+        if host and config or (not host and not config):
+            raise SmartDeviceException(
+                "One of host or config must be provded and not both"
+            )
+        if host:
+            config = DeviceConfig(host=host)
+        return await connect(config=config)  # type: ignore[arg-type]
```

Comment:
I think we should just pass the arguments to the factory and let it handle the merging config&host or erroring out like done here. This would just be a helper that calls the factory function and does nothing else.

Reply:
Done